app/assets/javascripts/components/campaign/campaign.jsx

```diff
@@ -0,0 +1,52 @@
+// /* eslint no-undef: 2 */
+import { Route, Switch } from 'react-router-dom';
+import { withRouter } from 'react-router';
+import React from 'react';
+import createReactClass from 'create-react-class';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import CampaignNavbar from '../common/campaign_navbar.jsx';
+import { getCampaign } from '../../actions/campaign1_actions';
+import CampaignAlerts from '../alerts/campaign_alerts.jsx';
+import CampaignOresPlot from './campaign_ores_plot.jsx';
+
+export const Campaign = createReactClass({
+  displayName: 'Campaign',
+
+  propTypes: {
+    campaign: PropTypes.object.isRequired,
+    match: PropTypes.object,
+  },
+
+  componentDidMount() {
+    const campaignSlug = this.props.match.params.slug;
+    return this.props.getCampaign(campaignSlug);
+  },
+
+  render() {
+    if (!this.props.campaign.uploads_in_use_count) {
```

Comment:
I believe the specs are still failing because of this... if the uploads_in_use_count value is 0, that's still a falsy value so this will skip rendering even after it receives the 0 value from the server.

You can do something like `if (this.props.campaign.uploads_in_use_count === '')` instead, so that will only skip rendering in the case of the initial empty string state, and any integer received from the server will allow it to render.

(A better approach here would be to add a `loading` attribute to the reducer, with `true` as the default, and then set it to false once the campaign data is received. But I think keeping it simpler for now is fine.)